tests/integration/high_availability/test_upgrade.py

```diff
@@ -124,7 +120,16 @@ async def test_upgrade_from_edge(ops_test: OpsTest, continuous_writes) -> None:
     assert leader_unit is not None, "No leader unit found"
 
     logger.info("Resume upgrade")
-    await juju_.run_action(leader_unit, "resume-upgrade")
+    while get_sts_partition(ops_test, MYSQL_APP_NAME) == 2:
+        # resume action sometime fails in CI, no clear reason
```

Comment:
not happy to have this workaround

Reply:
Me neither, but I should be clearer here, since I know the reason.
the run action recently changed behavior, raising when the action fails.
The action will fail when we the leader unit is the one target as the next one to be upgrade, e.g. when we have three units (0, 1*, 2) when resuming upgrade after unit 2 is completed, `resume-upgrade` will run on unit 1(leader) and set the partition to `1`, immediately rolling the pod, before the charm is able to respond the action hook